src/core/utils.cpp

```diff
@@ -497,6 +497,16 @@ bool isFunctionGloballyExcluded(Function *F) {
   return is_contained(Config.GlobalFunctionExclude, F->getName());
 }
 
+bool isCoroutine(Function *F) {
+  for (const auto &BB : *F)
+    for (const Instruction &I : BB)
+      if (auto *CI = dyn_cast<CallInst>(&I))
+        if (Function *F = CI->getCalledFunction())
+          if (F->getName().starts_with("llvm.coro.begin"))
+            return true;
+  return false;
```

Comment:
```suggestion
  for (const Instruction &I : instructions(F))
    if (isa<CoroBeginInst>(&I))
      return true;
  return false;
```

Reply:
Then we can go:
```cpp
  for (const Instruction &I : instructions(F))
    if (const auto *II = dyn_cast<IntrinsicInst>(&I);
        II && II->getIntrinsicID() == Intrinsic::coro_begin)
      return true;
  return false;
```
I think we should skip all coroutine intrinsics, esp. for the logic within BBDuplicate.